vms/avm/wallet_service_backend.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
```

Comment:
I'd go for the P-chain structure.
In `platformvm` we used to have a `txs/builder` package, which now has been renamed to `txs/txstest` since those builders are only used in tests.
Here I'd keep a `txs/builder` package as long as these builder are used in the APIs